src/app/components/budget-inputs/balance.service.ts

```diff
@@ -0,0 +1,100 @@
+import { Injectable, SimpleChanges } from '@angular/core';
+import { signal } from '@angular/core';
+import { WritableSignal } from '@angular/core';
+export interface Transaction {
+  amount: number;
+  description: string;
+}
+
+@Injectable({ providedIn: 'root' })
+export class BalanceService {
+  private balance = signal(0);
+  private incomes: WritableSignal<Transaction[]> = signal([]);
+  private expenses: WritableSignal<Transaction[]> = signal([]);
+
+  constructor() {
+    this.loadBalance();
+    this.loadTransactions();
+  }
+
+  getBalance = () => {
+    return this.balance();
+  };
+
+  getIncomes = () => {
+    return this.incomes();
+  };
+  getExpenses = () => {
+    return this.expenses();
+  };
+
+  addIncome = (amount: number, description: string) => {
+    this.incomes.update((incomes) => [
+      ...incomes,
+      { amount: amount, description: description },
+    ]);
+    this.balance.update((balance) => balance + amount);
+    this.saveBalance();
+    this.saveTransactions();
+  };
+
+  addExpense = (amount: number, description: string) => {
+    this.expenses.update((incomes) => [
+      ...incomes,
+      { amount: amount, description: description },
+    ]);
+    this.balance.update((balance) => balance + amount);
+    this.saveBalance();
+    this.saveTransactions();
+  };
+
+  saveBalance = () => {
+    localStorage.setItem('balance', this.balance().toString());
+  };
+  saveTransactions = () => {
```

Comment:
I really like that you are using some persistent layer. When I'm reviewing your PRs it really helps me to focus on the code and not on the setup.
Good Job, Tea!